example-schema/types/MainType.yaml

```diff
@@ -22,6 +22,11 @@ properties:
     type: Word
     description: Word description.
     label: Word label
+  someStringList:
+    type: Word
+    hasMany: true
```

Comment:
I am tempted to disagree with the use of `hasMany` here, because it implies **distinct** elements. We've been using it to describe the cardinality of relationships (this implies distinct elements), rather than to distinguish between primitive vs reference types. What about using a syntax like `Word[]` or `Array<Enum>`?

Reply:
I did initially have 
```
type: 
  - Word
```

In the end I preferred hasMany because it introduces no new concepts. Having hand held a number of pull requests into the schema my instinct is that having multiple different ways of expressing the cardinality of a field will make things more confusing. 

From a maintenance point of view, it also keeps our code simpler if all we have to check is a boolean property vs checking for some sort of pattern/structure in the type. I suspect that even if we went for using a patten/structure for authoring the cardinality, we'd probably end up converting it into a boolean internally, and also having to store the primitive type/enum type in a new property too

i.e.

```
    def.hasMany = /[]$/.test(def.type)
    def.realType = def.type.replace(/[]$/, '')
```